packages/runtime/src/widgets/Search.tsx

```diff
@@ -132,74 +119,57 @@ export const Search: React.FC<SearchProps> = ({
 
   const handleSelect = useCallback(
     (selectedValue: unknown): void => {
-      const selectedData = options.find(
-        ({ value: optionValue }) => optionValue === selectedValue,
-      )?.data;
-      setValue(selectedData);
-      onSelectAction?.callback({ value: selectedData });
+      setValue(selectedValue);
```

Comment:
the key shouldn't be renamed and it should set the complete object instead of a single value.